site/zenodo_rdm/migrator/transform/records.py

```diff
@@ -0,0 +1,229 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2022-2023 CERN.
+#
+# ZenodoRDM is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Zenodo migrator records transformers."""
+
+from invenio_rdm_migrator.streams.records import RDMRecordEntry, RDMRecordTransform
+from nameparser import HumanName
+
+
+class ZenodoRecordEntry(RDMRecordEntry):
+    """Transform Zenodo record to RDM record."""
+
+    def _created(self, entry):
+        """Returns the creation date of the record."""
+        return entry["created"]
+
+    def _updated(self, entry):
+        """Returns the creation date of the record."""
+        return entry["updated"]
+
+    def _version_id(self, entry):
+        """Returns the version id of the record."""
+        return entry["version_id"]
+
+    def _index(self, entry):
+        """Returns the index of the record."""
+        return entry.get("index", 0) + 1  # in legacy we start at 0
+
+    def _recid(self, entry):
+        """Returns the recid of the record."""
+        return str(entry["json"]["recid"])
+
+    def _pids(self, entry):
+        record = entry["json"]
+        r = {
+            "oai": {
+                "provider": "oai",
+                "identifier": record["_oai"]["id"],
+            },
+        }
+        if record.get("doi"):
+            r["doi"] = {
+                "client": "datacite",
+                "provider": "datacite",
+                "identifier": record["doi"],
+            }
+
+        return r
+
+    def _files(self, entry):
+        """Transform the files of a record."""
+        return {"enabled": True}
+
+    def _access(self, entry):
+        record = entry["json"]
+        is_open = record["access_right"] == "open"
+        r = {
+            "record": "public",
+            "files": "public" if is_open else "restricted",
+        }
+        if record["access_right"] == "embargoed":
+            r["embargo"] = {
+                "until": record["embargo_date"],
+                "active": True,
+            }
+        return r
+
+    def _metadata(self, entry):
```

Comment:
note: this looks like could be extrated to something like `MetadataEntry` class. However, out of scope for this PR.

Reply:
it would be nice to address also this https://github.com/inveniosoftware/invenio-rdm-migrator/issues/36

Reply:
@zzacharo if it is within the scope I can do it while tackling now the inclusion requests. Should not take too much time I think, no?

Reply:
it shouldn't but it has some impact of course in cds-rdm...but I think it will clean up the interface and improve readability :)